esp-hal/src/uart.rs

```diff
@@ -1582,7 +1575,15 @@ where
     }
 
     fn flush(&mut self) -> Result<(), Self::Error> {
-        self.tx.flush()
+        loop {
```

Comment:
`embedded_io::Write::flush(&mut self.tx)` would safe us from duplicating this code